### PR TITLE
moar google analytics

### PR DESCRIPTION
**Pull request by danfruehauf**
Addresses some of the google analytics metrics we're lacking. See also #1878.

### CODE REVIEW

**Comment by dnahodil** [comment]
I'll review

**Comment by dnahodil** [comment]
@jkburges, I've written a bunch of code for this PR now, too. So it probably wants a third person to run their eyes over it.

File: src/test/javascript/portal/cart/DownloadPanelSpec.js [hunk 1]
```diff
@@ -81,11 +82,18 @@ describe("Portal.cart.DownloadPanel", function() {
     });
 
     describe('clear all', function() {
-        it('calls to active geo network record store remove all', function() {
+        beforeEach(function() {
             window.setViewPortTab = jasmine.createSpy();
             downloadPanel._clearAllAndReset();
+        });
+
+        it('calls to active geo network record store remove all', function() {
```
Comment (jkburges) [inline src/test/javascript/portal/cart/DownloadPanelSpec.js]: The name of this `it` is outdated.

**Comment by jkburges** [comment]
In general, I think it's worth investigating re-jigging our GA code alongs the lines of:

* intercept all events
* filter out, by originating component and event type, the ones that we want to send to GA
* one function per component of interest, that can generate an object to send to GA from itself and the event

The benefit of this is that we can more easily add in new GAs, and see in one place what we're collecting.  We also don't end up with GA code intermixed with the other business/UI logic.

**Comment by dnahodil** [comment]
Having just done this work I think that idea has real merit. Maybe before we do more of this backlog item we should try a PoC and see how it might look.

**Comment by dnahodil** [comment]
I've added a new ticket on the board for someone to have a crack at that approach.

**Pull request merged**